common/src/com/thoughtworks/go/config/materials/Materials.java

```diff
@@ -140,9 +134,9 @@ public void cleanUp(File baseFolder, ConsoleOutputStreamConsumer consumer) {
         return allowed;
     }
 
-    boolean hasOneMaterialUseBaseFolder() {
+    public boolean hasOneMaterialUseBaseFolder() {
         for (Material material : this) {
-            if (material.getFolder() == null) {
+            if (material.getFolder() == null && (material instanceof ScmMaterial || material instanceof PluggableSCMMaterial)) {
```

Comment:
Shouldn't this logic be inside material?
e.g ```material.isSCM```
I'd even push ```material.getFolder() == null``` into a method ```material.hasDestinationFolder()``` if that's what the check is implying.